addons/crm_mail_plugin/tests/test_crm_mail_plugin.py

```diff
@@ -54,3 +39,82 @@ def test_get_contact_data(self):
             msg="The first lead belongs to the first partner, it should be returned")
         self.assertFalse([lead for lead in result["leads"] if lead["lead_id"] == lead_2.id],
             msg="The second lead does not belong to the first partner, it should not be returned")
+
+    @mock_auth_method_outlook('employee')
+    def test_crm_lead_create_multi_company(self):
+        """ Test that creating a record using the mail plugin for a contact belonging to a different company than the
+            default company of the user does not result in any issues.
+        """
+        # create Company_A and Company_B
+        company_a, company_b = self.env['res.company'].create([
+            {'name': 'Company_A'},
+            {'name': 'Company_B'},
+        ])
+
+        # create contact belonging to Company_B
+        contact = self.env['res.partner'].create({
+            'name': 'John Doe',
+            'email': 'john.doe@example.com',
+            'company_id': company_b.id,
+        })
+
+        # set default company to Company_A
+        self.env.user.company_id = company_a.id
+
+        # Set up the user's access to crm.lead
+        self.user_test.groups_id |= self.env.ref('sales_team.group_sale_salesman_all_leads')
+
+        # Add company_B to user_test to have access to records related to company_B
+        self.user_test.write({'company_ids': [(4, company_b.id)]})
+
+        # send request to create lead for the contact
+
+        params = {
+            'partner_id': contact.id,
+            'email_body': 'test body',
+            'email_subject': 'test subject',
+        }
+
+        result = self._make_rpc_call('/mail_plugin/lead/create', params)
+
+        # Check that the created lead record has the correct company and return the lead_id
+        self.assertIn(
+            'lead_id',
+            result,
+            msg='The lead_id should be returned in the response',
+        )
+
+        created_lead = self.env['crm.lead'].browse(result['lead_id'])
+
+        self.assertEqual(
+            created_lead.company_id,
+            company_b,
+            msg='The created record should belong to company_B',
+        )
+
+    def _make_rpc_call(self, url, params):
+        """
+        Makes an RPC call to the specified URL with the given parameters, and returns the 'result' key from the response.
+
+        :param params: A dictionary containing the parameters to send in the RPC call.
+        :type params: dict
+        :param url: The URL to send the RPC call to.
+        :type url: str
+        :return: The 'result' key from the response.
+        :rtype: dict
```

Comment:
a bit more "Odoo style"

```suggestion
        :param params (dict): A dictionary containing the parameters to send in the RPC call.
        :param url (str): The URL to send the RPC call to.
        :return (dict): The 'result' key from the response.